astropy/coordinates/tests/test_angles.py

```diff
@@ -929,12 +912,15 @@ def test_empty_sep():
 def test_create_tuple():
     """
     Tests creation of an angle with an (h,m,s) tuple
-
-    (d, m, s) tuples are not tested because of sign ambiguity issues (#13162)
     """
-    with pytest.warns(AstropyDeprecationWarning, match="hms_to_hours"):
-        a1 = Angle((1, 30, 0), unit=u.hourangle)
-    assert a1.value == 1.5
+    with pytest.raises(TypeError, match="no longer supported"):
+        Angle((1, 30, 0), unit=u.hourangle)
+    with pytest.raises(TypeError, match="no longer supported"):
+        Longitude((56, 14, 52.52))
+    with pytest.raises(TypeError, match="no longer supported"):
+        Longitude((12, 14, 52))  # ambiguous w/o units
+    with pytest.raises(TypeError, match="no longer supported"):
+        Longitude((12, 14, 52), unit=u.hour)
```

Comment:
The second and third asserts are very similar so we don't need both. Furthermore, [parametrizing the test](https://docs.pytest.org/en/stable/how-to/parametrize.html) can reduce code duplication.

Reply:
Parametrize how? You mean you want to split Longitude into a new test function and then parametrize that? I think it probably would result in the same LOC count anyway. Please clarify. Thanks!

Reply:
Something like this:
```python
@pytest.mark.parametrize("angle_class", [Angle, Longitude])
@pytest.mark.parametrize("unit", [u.hourangle, u.hour, None])
def test_create_tuple_fail(angle_class, unit):
    """Creating an angle from an (h,m,s) tuple should fail."""
    with pytest.raises(TypeError, match="no longer supported"):
        angle_class((12, 14, 52), unit=unit)
```

Reply:
Done. Thanks!